numpy/f2py/utils.py

```diff
@@ -0,0 +1,29 @@
+"""Global f2py utilities."""
+
+import os
+import contextlib
+import tempfile
+import pathlib
+import shutil
+
+def get_f2py_dir():
+	"""Return the directory where f2py is installed."""
+	return os.path.dirname(os.path.abspath(__file__))
+
+@contextlib.contextmanager
+def open_build_dir(build_dir: list or None, compile: bool):
```

Comment:
Nit: in Python, `list or None` always evaluates to `list`. Use `Optional[list]` or `Union[list, type(None)]`?

Reply:
Thanks! I didn't know that. Making changes.

Reply:
Made some changes, please have a look.